### ISSUE TITLE
UDP Support; waiting on test infrastructure

### ISSUE THREAD

**Issue by KrisNey-MSFT** [sonic-net/DASH#89]
## Related information
Scope - P4 DASH pipeline. Requires definition of UDP header in overlay pipeline headers, proper parsing, and match on L4 ports along with TCP ports.

### Notes

- Needs to be coded in P4 behavioral model
- No change in bmv2, just P4

**Comment by venkat-pullela-keys** [comment]
@marian-pritsak, assumed that all the layers of headers in packet transforms document are handled eventually. Where the  customer TCP flow is looked up in connection tracking, UDP can tag along by using common src/dst ports.

**Comment by KrisNey-MSFT** [comment]
Per @marian-pritsak UDP src/dst port s/be changed to flow port.  Need to ask how we go about ICMP (w/o src port) and ACLs in Community meeting.  @GeraldatMicrosoft and @KrisNey-MSFT to investigate.  Need a state machine (w/state criteria) to close UDP flows, Venkat has some ideas.  For UDP we could do flow lengths, when the gap extends past a point we end the flow (similarly to an idle timeout).

**Comment by KrisNey-MSFT** [comment]
UDP support -done with code. Waiting for testing infrastructure to commit the code. Will wait this week also, otherwise I will go ahead and commit.

**Comment by KrisNey-MSFT** [comment]
Header Unions
code for XN tracking (stateful P4) they are already using the header union
Want to integrate into UDP support, but keep separate - create separate Issue however wait for Marian

**Comment by yusefMS06** [comment]
@venkat-pullela-keys @marian-pritsak Can we mark as unblocked?

**Comment by ashutosh-agrawal** [comment]
@KrisNey-MSFT @mgheorghe Can we close this issue if the test infrastructure supports UDP already. Also, as listed in the original notes for this issue, it was a P4 limitation and not bmv2.

**Comment by KrisNey-MSFT** [comment]
UDP header s/be there; @venkat-pullela-keys could you confirm if anything else is needed?

**Comment by KrisNey-MSFT** [comment]
@venkat-pullela-keys - can we determine that this PR is closed, or confirm whether more is needed?
TY, @KrisNey-MSFT

**Comment by KrisNey-MSFT** [comment]
We believe UDP is supported now, @venkat-pullela-keys can you check to see if anything is missing?  Header parsing metadata, once this is done, the flow creation/pipeline matching s/be in place.
